### PR TITLE
Alena burak/6 3 coach check 2

### PR DESCRIPTION
**Pull request by ElenaBurak**

### CODE REVIEW

**Review by vlad12345123 — APPROVED** [review]

File: exalearn-api/ExaLearn.Bl/Services/TestService.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            _userRepository = userRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<PassedTestForCheckDTO> GetUserTestByPassedTestIdAsync(int passedTestId, string checker)
+        {
+            var passedTest = await _passedTestRepository.GetByIdAsync(passedTestId);
+            var checkerId = await _userRepository.FindByNameAsync(checker);
+            passedTest.CheckerId = checkerId;
+            passedTest.Status = Shared.Enums.StatusType.InCoachProgress;
```
Comment (vlad12345123) [inline exalearn-api/ExaLearn.Bl/Services/TestService.cs]: add using Shared.Enums.StatusType;

File: exalearn-api/ExaLearn.Dal/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -20,5 +20,12 @@ public async Task<string> GetUserRole(int id)
 
             return role.Name;
         }
+
+        public async Task<int> FindByNameAsync(string userName)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/UserRepository.cs:24]: ????????????????

File: exalearn-api/ExaLearn.Tests/Services/UserAnswerServiceTests.cs [hunk 3]
```diff
@@ -15,17 +16,19 @@ namespace ExaLearn.Tests.Services
...
         private readonly IUserAnswerService _userAnswerService;
 
         public UserAnswerServiceTests()
         {
-            _mockUserAnswerRepository = new Mock<IUserAnswerRepository>();
+            _mockUserAnswerRepository = new Mock<IUserAnswerRepository>();         
 
             _mapper = MapperConfigurationProvider.GetConfig().CreateMapper();
 
-            _userAnswerService = new UserAnswerService(_mockUserAnswerRepository.Object, _mapper);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserAnswerServiceTests.cs:28]: ????

File: exalearn-api/ExaLearn.WebApi/Controllers/TestController.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+        public TestController(ITestService testService)
+        {
+            _testService = testService;
+        }
+
+        [HttpGet("getTestForCoachCheck/{passedTestId}")]
+        public async Task<IActionResult> GetTestForCoachCheck(int passedTestId)
+        {
+            var checker = User.Identity.Name;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/TestController.cs]: email

**Review by romanstat — DISMISSED** [review]

**Comment by artemgael96** [comment]
confilcts

**Pull request closed without merging**